web/src/app/components/header/header.component.html

```diff
@@ -54,14 +54,16 @@
         <button
           mat-flat-button
           class="finish-edit-button"
-          [disabled]="!isDraftSurveyDirtyAndValid()"
+          [disabled]="publishingChanges || !isDraftSurveyDirtyAndValid()"
```

Comment:
nit: can we rename this to `isPublishingChanges` just to make it more clear that this variable represents a state